tensorflow/compiler/xla/service/plaidml/compiler.cc

```diff
@@ -434,8 +617,37 @@ StatusOr<std::unique_ptr<Program>> PlaidMLCompiler::ProgramFromHloModule (
           // Tensor reshape operation
           auto op = ::plaidml::edsl::reshape(instr_map[operand_ids[0]], dims);
           instr_map.insert(std::make_pair(cur_instr_id, op));
-          //program_str_cpp += tabs + "std::vector<int64_t> shape" + unique_name + " = " + dims + ";\n";
-          //program_str_cpp += tabs + "auto " + unique_name + " = reshape(" + operand_names[0] + ", shape" + unique_name + ");\n";
+          break;
+        }
+        case HloOpcode::kReverse: {
+          // Reverse in XLA is Flip in Numpy/Op Lib
+          auto op = instr_map[operand_ids[0]];
+          auto instr_dims = instruction->dimensions();
+          for (auto dim : instr_dims) {
+            // Can only flip on one axis at a time
+            VLOG(2) << "Performing reverse on dimension " << dim;
+            op = plaidml_op::flip(op, dim);
+          }
+          instr_map.insert(std::make_pair(cur_instr_id, op));
+          break;
+        }
+        case HloOpcode::kSelect: {
+          // Tensor select conditional operation
+          auto op = ::plaidml::edsl::select(instr_map[operand_ids[0]], instr_map[operand_ids[1]], instr_map[operand_ids[2]]);
+          instr_map.insert(std::make_pair(cur_instr_id, op));
+          break;
+        }
+        case HloOpcode::kSlice: {
+          // Tensor slice operation
+          // Grab start points, end points, and strides
+          std::vector<Value> dims_slices;
+          for (auto i = 0; i < dims.size(); i++) {
+            VLOG(2) << "Slicing dimension " << i << " starting at index " << instruction->slice_starts(i) << " ending at index " << instruction->slice_limits(i) << " with stride " << instruction->slice_strides(i);
+            dims_slices.push_back(::plaidml::edsl::make_tuple(instruction->slice_starts(i), instruction->slice_limits(i), instruction->slice_strides(i)));
+          }
+          auto args = ::plaidml::edsl::make_tuple(instr_map[operand_ids[0]], dims_slices);
+          auto op = plaidml_op::details::op("slice", args).as_tensor();
```

Comment:
Why are we using `details` directly? We should just use `plaidml::op::slice()`.

Reply:
We need to do strided slices, which the current `slice` API doesn't yet have support for (we discussed this in a Tech Discussions meeting a couple of weeks ago). I'm planning on making a PR soon to add support for strided slicing, and then we'll be able to use `plaidml::op::slice()`

Reply:
FYI: the `details` namespace is not considered to be part of the public API and may change at any time.